src/corelibs/U2Formats/src/FastaFormat.cpp

```diff
@@ -70,180 +54,184 @@ FastaFormat::FastaFormat(QObject *p)
                            "The format also allows for sequence names and comments to precede the sequences.");
 }
 
-static QVariantMap analyzeRawData(const QByteArray &data) {
-    int hasGaps = false;
+/** Returns FormatDetectionResult properties (hints) for the given data. */
+static QVariantMap buildFormatDetectionHints(const QString &data) {
+    bool hasGaps = false;
     int minLen = -1;
     int maxLen = -1;
-    int len = 0;
+    int currentSequenceLength = 0;
     int nSequences = 0;
 
-    QTextStream input(data, QIODevice::ReadOnly);
-    QString line;
+    QString rwData = data;
+    QTextStream textStream(&rwData, QIODevice::ReadOnly);
     do {
-        line = input.readLine();
-        if (line[0] == FastaFormat::FASTA_HEADER_START_SYMBOL) {
+        QString line = textStream.readLine();
+        if (line.startsWith(FastaFormat::FASTA_HEADER_START_SYMBOL)) {
             nSequences++;
-            if (len > 0) {
-                minLen = minLen == -1 ? len : qMin(minLen, len);
-                maxLen = maxLen == -1 ? len : qMax(maxLen, len);
+            if (currentSequenceLength > 0) {
+                minLen = minLen == -1 ? currentSequenceLength : qMin(minLen, currentSequenceLength);
+                maxLen = maxLen == -1 ? currentSequenceLength : qMax(maxLen, currentSequenceLength);
             }
-            len = 0;
+            currentSequenceLength = 0;
         } else {
-            len += line.length();
+            currentSequenceLength += line.length();
             if (!hasGaps && line.contains(U2Msa::GAP_CHAR)) {
                 hasGaps = true;
             }
         }
-    } while (!input.atEnd());
+    } while (!textStream.atEnd());
 
-    QVariantMap res;
-    res[RawDataCheckResult_Sequence] = true;
+    QVariantMap hints;
+    hints[RawDataCheckResult_Sequence] = true;
     if (hasGaps) {
-        res[RawDataCheckResult_SequenceWithGaps] = true;
+        hints[RawDataCheckResult_SequenceWithGaps] = true;
     }
     if (nSequences > 1) {
-        res[RawDataCheckResult_MultipleSequences] = true;
+        hints[RawDataCheckResult_MultipleSequences] = true;
     }
     if (minLen > 0) {
-        res[RawDataCheckResult_MaxSequenceSize] = maxLen;
-        res[RawDataCheckResult_MinSequenceSize] = minLen;
+        hints[RawDataCheckResult_MaxSequenceSize] = maxLen;
+        hints[RawDataCheckResult_MinSequenceSize] = minLen;
     }
-    return res;
+    return hints;
 }
 
-FormatCheckResult FastaFormat::checkRawTextData(const QByteArray &rawData, const GUrl &) const {
-    const char *data = rawData.constData();
-    int size = rawData.size();
-
-    int n = TextUtils::skip(TextUtils::WHITES, data, size);
-    int newSize = size - n;
-    const char *newData = data + n;
-    if (newSize <= 0 || (newData[0] != FASTA_HEADER_START_SYMBOL && newData[0] != FASTA_COMMENT_START_SYMBOL)) {
+FormatCheckResult FastaFormat::checkRawTextData(const QString &dataPrefix, const GUrl &) const {
```

Comment:
You rewrote this function and several checks are gone now (e.g `newSize <= 0` or `TextUtils::contains(TextUtils::BINARY, data, size)`). Maybe it's better to get them back?

Reply:
TextDocumentFormat methods should never be called for binary data now. The check is done in the base class before any child class method is called.